src/components/SingleDatePicker.jsx

```diff
@@ -188,14 +273,35 @@ export default class SingleDatePicker extends React.Component {
   }
 
   onDayMouseEnter(day) {
+    if (this.isTouchDevice) return;
+    const { hoverDate, visibleDays } = this.state;
+
+    let modifiers = {};
+    modifiers = this.deleteModifier(modifiers, hoverDate, 'hovered');
+    modifiers = this.addModifier(modifiers, day, 'hovered');
+
     this.setState({
       hoverDate: day,
+      visibleDays: {
+        ...visibleDays,
+        ...modifiers,
+      },
     });
   }
 
   onDayMouseLeave() {
+    const { hoverDate, visibleDays } = this.state;
+    if (this.isTouchDevice || !hoverDate) return;
+
+    let modifiers = {};
+    modifiers = this.deleteModifier(modifiers, hoverDate, 'hovered');
```

Comment:
The empty object is so that when we call `addModifier`/`deleteModifier` the effects stack rather than overwriting each other. So for instance if something happens that causes two different modifiers to be added, we spread everything onto this object and then set the state at the end.